lib/utils.ts

```diff
@@ -2,14 +2,18 @@ import { SALEOR_DOMAIN_HEADER } from "@saleor/app-sdk/const";
 import { Request } from "retes";
 
 import { FetchAppDetailsDocument } from "../generated/graphql";
-import { getEnvVars } from "./environment";
 import { createClient } from "./graphql";
+import { apl } from "./saleorApp";
 
-export async function getAppIdFromApi(request: Request) {
+export async function getAppIdFromApi(request: Request): Promise<string | undefined> {
   // Get installed App ID from the Saleor API based on request
   const { [SALEOR_DOMAIN_HEADER]: saleorDomain } = request.headers;
+  const authData = await apl.get(saleorDomain);
+  if (!authData) {
+    return undefined;
+  }
   const client = createClient(`https://${saleorDomain}/graphql/`, async () =>
-    Promise.resolve({ token: (await getEnvVars()).SALEOR_AUTH_TOKEN })
+    Promise.resolve({ token: authData?.token })
   );
```

Comment:
shuldn't it be in app-sdk? or even rewrite to middleware and add it to req context?

Reply:
I wonder if we should even promote using separate middlewares for each validation, maybe instead we should have something like:
- withSaleorEventValidated - it would be used for webhooks from Saleor: check event header, payload and JWT
- withDashboardRequestValidated - for app requests from the dashboard: check headers and JWT

We would hide separate checks from the devs, exposing easier functions to work with. WDYT?
It would help us to not check for the same headers multiple times and minimize the amount of Auth API calls. 

Adding Auth Data to the ctx seems like a good idea, it's common to use this data in the handlers

Reply:
Yes, IMO it should be a single method with a config. We don't have to do this in this PR. But some example we can think of:

```
// api/webhooks/order-crated.ts

export default createSaleorWebhookHandler({
  webhookName: 'ORDER_CREATED',
  handler(req, res, ...){ /* domain logic */},
  ... // other options, grow in time
})


```